src/ml/dimensionality_reduction/pca.nim

```diff
@@ -26,3 +29,19 @@ proc pca*[T: SomeReal](x: Tensor[T], nb_components = 2): Tensor[T] =
 
   # [Nb_obs, Nb_feats] * [Nb_feats, Nb_components], don't forget to reorder component in descending order
   result = mean_centered * eigvecs[_, ^1..0|-1]
+
+  if transform:
+    return result
+  else:
+    return eigvecs
+
+proc pca*[T: SomeReal](x: Tensor[T], principal_axes: Tensor[T]): Tensor[T] =
+  ## Principal Component Analysis (PCA) projection
+  ## Inputs:
+  ##    - A matrix of shape [Nb of observations, Nb of features]
+  ##    - A matrix of shape [Nb of observations, Nb of principal axes] to project on
+  ##
+  ## Returns:
+  ##    - A matrix of shape [Nb of observations, Nb of components]
+  let mean_centered = x .- x.mean(axis=0)
+  result = mean_centered * principal_axes[_, ^1..0|-1]
```

Comment:
As we switch to descending order, `principal_axes` instead of `principal_axes[_, ^1..0|-1]`